site/source/docs/porting/connecting_cpp_and_javascript/Interacting-with-code.rst

```diff
@@ -622,21 +622,46 @@ See `test_add_function in test/test_core.py`_ for an example.
 You should build with ``-sALLOW_TABLE_GROWTH`` to allow new functions to be
 added to the table. Otherwise by default the table has a fixed size.
 
-.. note:: When using ``addFunction`` on LLVM Wasm backend, you need to provide
-   an additional second argument, a Wasm function signature string. Each
-   character within a signature string represents a type. The first character
-   represents the return type of a function, and remaining characters are for
-   parameter types.
+When using ``addFunction`` on the LLVM Wasm backend, you need to provide
+an additional second argument, a Wasm function signature string, explained below.
+See `test/interop/test_add_function_post.js <https://github.com/emscripten-core/emscripten/blob/main/test/interop/test_add_function_post.js>`_ for an example.
+
+
+.. _interacting-with-code-function-signatures:
+
+Function Signatures
+===================
+
+The LLVM Wasm backend requires a Wasm function signature string when using ``addFunction``
+and in JavaScript libraries. Each character within a signature string represents a type. The first character
+represents the return type of a function, and remaining characters are for
+parameter types.
 
    - ``'v'``: void type
    - ``'i'``: 32-bit integer type
-   - ``'j'``: 64-bit integer type (currently does not exist in JavaScript)
+   - ``'j'``: 64-bit integer type (see note below)
    - ``'f'``: 32-bit float type
    - ``'d'``: 64-bit float type
+   - ``'p'``: 32-bit or 64-bit pointer (MEMORY64)
+
+For example, if you add a function that takes an integer and does not return
+anything, the signature is ``'vi'``.
+
+``'j'`` must be used with the ``__i53abi`` decorator.  It simplifies passing 64-bit numbers in JavaScript libraries.  It cannot be used with ``addFunction``.  Here is an example of a library function that sets the size of a file using a 64-bit value and returns
```

Comment:
When `j` is used there are several ways in which the parameter value will be passed to JavaScript.    By default the value will either be passed as a single bigint or a pair of JavaScript numbers (double) depending on whether  the `WASM_BIGINT` settings is enabled.  In addition, if you only require 53bits of precision you can add the `__i53abi` decorator, which will ignore the upper bits of the `i64` and the value will be recieved as a single JavaScript number (double).